pkg/cmd/pulumi/deployment.go

```diff
@@ -66,6 +67,87 @@ func newDeploymentCmd() *cobra.Command {
 		"Override the file name where the deployment settings are specified. Default is Pulumi.[stack].deploy.yaml")
 
 	cmd.AddCommand(newDeploymentSettingsCmd())
+	cmd.AddCommand(newDeploymentRunCmd())
+
+	return cmd
+}
+
+func newDeploymentRunCmd() *cobra.Command {
+	// Flags for remote operations.
+	remoteArgs := RemoteArgs{}
+
+	var stack string
+	var jsonDisplay bool
+	var suppressPermalink string
+
+	cmd := &cobra.Command{
+		Use:   "run <operation> [url]",
+		Short: "Launch a deployment job on Pulumi Cloud",
+		Long:  "",
+		Args:  cmdutil.RangeArgs(1, 2),
+		Run: cmdutil.RunResultFunc(func(cmd *cobra.Command, args []string) result.Result {
+			ctx := cmd.Context()
+
+			operation, err := apitype.ParsePulumiOperation(args[0])
+			if err != nil {
+				return result.FromError(err)
+			}
+
+			var url string
+			if len(args) > 1 {
+				url = args[1]
+			}
+
+			display := display.Options{
+				Color:       cmdutil.GetGlobalColorization(),
+				JSONDisplay: jsonDisplay,
+				// we only suppress permalinks if the user passes true. the default is an empty string
+				// which we pass as 'false'
+				SuppressPermalink: suppressPermalink == "true",
+			}
+
+			project, _, err := readProject()
+			if err != nil && !errors.Is(err, workspace.ErrProjectNotFound) {
+				return result.FromError(err)
+			}
+
+			currentBe, err := currentBackend(ctx, project, display)
+			if err != nil {
+				return result.FromError(err)
+			}
+
+			if !currentBe.SupportsDeployments() {
+				return result.FromError(fmt.Errorf("backends of this type %q do not support deployments",
+					currentBe.Name()))
+			}
+
+			s, err := requireStack(ctx, stack, stackOfferNew|stackSetCurrent, display)
+			if err != nil {
+				return result.FromError(err)
+			}
+
+			if errResult := validateDeploymentFlags(url, remoteArgs); errResult != nil {
+				return errResult
+			}
+
+			return runDeployment(ctx, display, operation, s.Ref().FullyQualifiedName().String(), url, remoteArgs)
+		}),
+	}
+
+	// Remote flags
+	remoteArgs.applyFlagsForDeploymentCommand(cmd)
+
+	cmd.PersistentFlags().StringVar(
+		&suppressPermalink, "suppress-permalink", "",
+		"Suppress display of the state permalink")
+
+	cmd.PersistentFlags().StringVarP(
+		&stack, "stack", "s", "",
+		"The name of the stack to operate on. Defaults to the current stack")
+
+	cmd.Flags().BoolVarP(
+		&jsonDisplay, "json", "j", false,
+		"Serialize the update diffs, operations, and overall output as JSON")
```

Comment:
What does the json output look like?

Reply:
Actually it doesnt work, I will remove it